homeassistant/components/tesla_wall_connector/__init__.py

```diff
@@ -0,0 +1,184 @@
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import (
+    CONF_SCAN_INTERVAL_CHARGING,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_SCAN_INTERVAL_CHARGING,
+    DOMAIN,
+    WALLCONNECTOR_DATA_LIFETIME,
+    WALLCONNECTOR_DATA_VITALS,
+    WALLCONNECTOR_DEVICE_NAME,
+)
+
+PLATFORMS: list[str] = ["binary_sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Tesla Wall Connector from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    hostname = entry.data[CONF_HOST]
+    poll_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    poll_interval_charging = entry.options.get(
+        CONF_SCAN_INTERVAL_CHARGING, DEFAULT_SCAN_INTERVAL_CHARGING
+    )
+
+    wall_connector = WallConnector(host=hostname, session=async_get_clientsession(hass))
+
+    try:
+        version_data = await wall_connector.async_get_version()
+    except WallConnectorError as ex:
+        raise ConfigEntryNotReady from ex
+
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="tesla-wallconnector",
+        update_interval=timedelta(seconds=poll_interval),
+    )
+
+    async def async_update_data():
+        """Fetch new data from the Wall Connector."""
+
+        previous_contactor_closed = (
+            coordinator.data[WALLCONNECTOR_DATA_VITALS].contactor_closed
+            if coordinator.data is not None
+            and coordinator.data[WALLCONNECTOR_DATA_VITALS] is not None
+            else False
+        )
+
+        try:
+            vitals = await wall_connector.async_get_vitals()
+            lifetime = await wall_connector.async_get_lifetime()
+        except WallConnectorConnectionTimeoutError as ex:
+            raise UpdateFailed(
+                f"Could not fetch data from Tesla WallConnector at {hostname}: Timeout"
+            ) from ex
+        except WallConnectorConnectionError as ex:
+            raise UpdateFailed(
+                f"Could not fetch data from Tesla WallConnector at {hostname}: Cannot connect"
+            ) from ex
+        except WallConnectorError as ex:
+            raise UpdateFailed(
+                f"Could not fetch data from Tesla WallConnector at {hostname}: {ex}"
+            ) from ex
+
+        if previous_contactor_closed != vitals.contactor_closed:
+            coordinator.update_interval = (
+                timedelta(seconds=poll_interval)
+                if not vitals.contactor_closed
+                else timedelta(seconds=poll_interval_charging)
+            )
+            _LOGGER.debug(
+                "Contactor closed: %s. Update interval: %s",
+                str(vitals.contactor_closed),
+                str(coordinator.update_interval),
```

Comment:
`_LOGGER` should take care of stringifying these since use you using `%s`

Reply:
Fixed in 9ab5838